cou/commands.py

```diff
@@ -408,46 +411,31 @@ def prompt(self) -> bool:
         return not self.auto_approve
 
     @property
-    def is_hypervisors_command(self) -> bool:
-        """Whether if the command passed is specific to hypervisors.
+    def scope(self) -> UpgradeScope:
+        """Define which is the scope to upgrade.
 
-        :return: True if is hypervisors, false otherwise.
-        :rtype: bool
-        """
-        return self.upgrade_group == HYPERVISORS
+        Scopes can be control-plane, data-plane or the whole cloud.
 
-    @property
-    def is_data_plane_command(self) -> bool:
-        """Whether if the command passed is data-plane related.
-
-        :return: True if is data-plane, false otherwise.
-        :rtype: bool
+        :return: Scope to upgrade
+        :rtype: UpgradeScope
         """
-        return self.is_hypervisors_command or self.upgrade_group == DATA_PLANE
+        return UpgradeScope(self.upgrade_group)
 
-    @property
-    def is_control_plane_command(self) -> bool:
-        """Whether if the command passed is control-plane related.
 
-        :return: True if is control-plane, false otherwise.
-        :rtype: bool
-        """
-        return self.upgrade_group == CONTROL_PLANE
-
-    @property
-    def is_generic_command(self) -> bool:
-        """Whether the command has no specific group for upgrading.
+class UpgradeScope(Enum):
+    """Possible upgrade scopes."""
 
-        When this happens both control-plane and data-plane should generate
-        plans.
-        E.g of generic commands:
-            - cou upgrade
-            - cou plan
+    CONTROL_PLANE = CONTROL_PLANE
+    DATA_PLANE = DATA_PLANE, HYPERVISORS
+    WHOLE_CLOUD = None
 
-        :return: True if upgrade_group is None, false otherwise.
-        :rtype: bool
-        """
-        return self.upgrade_group is None
+    def __new__(cls, *values: Any) -> UpgradeScope:
```

Comment:
What is this? What is canonical value?

Reply:
This is a tweak to have multi-value enum. With this is possible to have something like:

```
class A(Enum):
  B = 1, "foo"
  C = 2, "bar"
```

This means that A(1) == A("foo") == A.B and A(2) == A("bar") == A.C

> What is canonical value?
It's the value as the Enum implementation normally works